Makefile

```diff
@@ -31,8 +34,9 @@ clean: stop
 	rm -rf logs/*
 	if [ -f airflow-worker.pid ]; then rm airflow-worker.pid; fi
 
-pip-install-local: pip-compile
-	pip install -r requirements.txt
+# Install Python packages in the local environment, but ignore pip's "Requirement already satisfied" messages.
+pip-install-local:
+	set -o pipefail; pip install -r requirements.txt -r requirements-dev.txt | { grep -v 'already satisfied' || test $$? = 1; }
```

Comment:
It's unfortunate that this kludge is necessary to ignore pip's "_Requirement already satisfied_" messages (see pypa/pip#5900), but IMO it's worth it so packages can be auto-installed when needed without being barraged by hundreds of "_Requirement already satisfied_" messages every time we run commands like `make fixes` or `make test`.

Reply:
It's not pretty but I think it's ok with the comment explaining it.

Is the `test $? = 1` so it still errors out if grep returns >1?  Seems overkill to handle a grep failure that's fine

Reply:
Correct.  grep will exit with status `1` if it didn't select any lines (I wish there was an option to override that behavior), or `>1` if an actual error occurred.  So we have to do something to prevent the result status of the overall command being non-zero when no packages needed to be installed (i.e. `pip` reported "_Requirement already satisfied_" for everything), because `make` would balk at that if the target was being run as a prerequisite.

A slightly simpler alternative would be `... | { grep -v 'already satisfied' || true; }`, which would ignore any potential grep errors, but since I'm not familiar with what errors those might be I figured I should play it safe.

Reply:
> This results in `/bin/sh: 1: set: Illegal option -o pipefail` on certain Linux distributions such as Ubuntu.

Good point. We could avoid that by [configuring](https://www.gnu.org/software/make/manual/html_node/Choosing-the-Shell.html) `make` to use `bash` instead of `sh`. However...

> I'm personally in favour of not having `pip-install-local` as requirement for `pip-compile`, `fixes`, and `test`. IMO having to install a project dependencies before working on the project is ubiquitous enough that we don't need to provide crutches in the `makefile` to enforce this order of operations. It also adds a few seconds of dependency verification before each test run which is annoying when iterating on tests. It also helps keeping a simpler make command.

Those are fair arguments.  Also, since `pip-install-local` is running `pip install` commands directly in the local system without guardrails like bigquery-etl uses to enforce a virtual environment it's actually a bit dangerous, as people could easily run those other commands without realizing they'll also run `pip install` and thus accidentally modify their global Python installation.  So I'll revert the change to have `pip-install-local` as a prerequisite for other commands.

Longer term, given the mostly containerized nature of the dev workflow here, IMO it would be great to fully containerize the dev workflow so the only thing that would need to be installed locally is Docker.